deepchecks/core/task_type.py

```diff
@@ -20,4 +20,5 @@ class TaskType(Enum):
     CLASSIFICATION = 'classification'
     OBJECT_DETECTION = 'object_detection'
     SEMANTIC_SEGMENTATION = 'semantic_segmentation'
+    TOKEN_CLASSIFICATION = 'token_classification'
```

Comment:
No reason for this to be shared. The fact that vision and NLP (and tabular) have tasks named "classification" does not mean these are the same.
What do you gain from using this as a shared enum? This is confusing in my opinion